wolfssl/wolfcrypt/chacha.h

```diff
@@ -57,6 +57,8 @@ enum {
 
 typedef struct ChaCha {
     word32 X[CHACHA_CHUNK_WORDS];           /* state of cipher */
+    word32 tmp[CHACHA_CHUNK_WORDS];         /* left over stream */
```

Comment:
Do we really need to nearly double the size of the struct?  Why not just use X[] and delay the plus one operation until all of X[] is consumed?

Reply:
I'll look at changing it, and add a commit shortly

Reply:
Looking at it more, before these PR changes the local variable "temp" in wc_Chacha_Process held the stream that was XOR'd with the input. It's going to be a trade off of performance versus size, to recreate "temp" on the next call, when not all of the data is used, an extra call to wc_Chacha_wordtobyte(temp, ctx->X) would need to be done in order to recreate the stream.

Reply:
Agreed, for a small set of users and small set of their data cases an extra function call is used.  But compare that to nearly doubling the size of the ChaCha struct for ALL users in all cases.  I think we should remove the doubling if easy to do correctly.